src/Pages/Page.php

```diff
@@ -108,4 +109,46 @@ public static function collectionDetails($key = null)
 
         return $key ? $names->$key : $names;
     }
+
+    public static function flattenForSelect()
```

Comment:
kunnen deze methods niet beter in een PageCollection, zoals de oorspronkelijke RelatedCollection? Of eventueel in de HasCollection trait.

Misschien moeten we dan ook de getRelationId(), getRelationLabel() enzo. veralgemenen zodat dit voor beide gevallen (relations en menubuilder) van toepassing kan zijn.